_posts/2020-02-17-eda-architecture.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: post
+title:  "An event driven architecture"
+excerpt: "Why and How we ditched our old architecture based on synchronous REST services for a completely asynchronous event driven architecture."
+date:   2020-02-07 12:53:00 +0100
+categories: architecture
+tags: python services rust broker event bus rabbitmq architecture
+---
+
+## Introduction
+
+At [Allo-Media](https://www.allo-media.net/en/), like many other businesses, our value chain looks like a pipeline: we collect conversations (mainly through phone) sent to us by our customers, we transcribe them, we tag the transcripts with named entities, we anonymize both the transcript and the audio, then we qualify the content with semantic tags, and finally we index them and provide a UI and API to consult, search, analyze the conversations. All those steps are completed automatically by NLP and AI algorithms.
```

Comment:
typo: a UI => an UI
nit: search, analyze => search and analyze

Reply:
I disagree. "An UI" is not natural to say while "a UI" is smooth.

> The rule states that “a” should be used before words that begin with consonants (e.g., b, c ,d) while “an” should be used before words that begin with vowels (e.g., a,e,i). Notice, however, that the usage is determined by the pronunciation and not by the spelling, as many people wrongly assume.
 — https://www.dailywritingtips.com/using-a-and-an-before-words/